doc/developer_guild.md

```diff
@@ -0,0 +1,471 @@
+# Extend Functionalities of GoTorch
+
+At the very core of PyTorch is *libtorch*, which is a C++ library that contains
+three levels of abstraction:
+
+1. *Native function*s are low-level functions, most of which operate tensors.
+   They are the basic mathematical build blocks of deep neural networks.
+1. *Functional*s are high-level functions built on native functions, each
+   functional may call several native functions.
+1. *Module*s are c++ classes/structs/templates that encapsulate multiple native
+   functions or functionals.
+
+Modules and functionals form the Pytorch C++ frontend.  AI practitioners can
+directly use libtorch modules or functionals to build their AI applications.
+
+libtorch contains about 1600 native functions, they are the foundations of
+PyTorch, GoTorch, and the PyTorch C++ frontend. In PyTorch, Python functions
+in package `torch.nn.functional` and modules classes in `torch.nn` call native
+functions via [`pybind`](https://github.com/pybind/pybind11).  In GoTorch, the
+corresponding packages `gotorch/nn/functional` and `gotorch/nn` call Go wrappers
+of native functions.
+
+The tutorial demonstrates how to wrap libtorch native functions or functionals
+into GoTorch using [Cgo](https://blog.golang.org/cgo), and how to build modules
+based on native functions and functionals.
+
+## Wrap PyTorch Native Functions
```

Comment:
We already have `wrap_native_functions.md`!

Reply:
Done.